src/main/scala/com/gu/scanamo/DerivedDynamoFormat.scala

```diff
@@ -55,3 +56,23 @@ trait DerivedDynamoFormat {
       def write(t: T): AttributeValue = formatR.value.write(gen.to(t))
     }
 }
+
+trait NoFormatError {
+  import scala.language.experimental.macros
+  implicit def moreHelpfulErrorMessage[T]: DynamoFormat[T] = macro MacroOfLastResort.moreHelpfulErrorMessage[T]
+}
+
+@bundle
+class MacroOfLastResort(val c: scala.reflect.macros.blackbox.Context) {
```

Comment:
My understanding is that it's essentially the state of compilation. So for instance it can give you the `enclosingPosition` which is the point of code from which the macro was called and can be used for things like aborting  macro.

It's also responsible for determining the compilation universe you're in, which I'm not going to pretend I understand well enough to give a coherent explanation of.